hd-docs/backend/collector-service.md

```
@@ -0,0 +1,364 @@
+async function asPdf({ fullFilePath, filename, options = {} }) {
+  const pdfLoader = new PDFLoader(fullFilePath, { splitPages: true });
+  let docs = await pdfLoader.load();
+  
+  // OCR fallback for image-based PDFs
+  if (docs.length === 0) {
+    console.log(`No text content found for ${filename}. Will attempt OCR parse.`);
+    docs = await new OCRLoader({
+      targetLanguages: options?.ocr?.langList,
+    }).ocrPDF(fullFilePath);
+  }
+  
```

markdownlint-cli2 (0.17.2)

Lists should be surrounded by blank lines

(MD032, blanks-around-lists)